packages/components/src/internal/app/utils.ts

```diff
@@ -222,57 +224,223 @@ export function getMenuSectionConfigs(user: User, currentApp: string): List<Map<
+                    [WORKFLOW_KEY]: workflowConfig,
+                    [MEDIA_KEY]: getMediaSectionConfig(appBase),
+                    [NOTEBOOKS_KEY]: getNotebooksSectionConfig(appBase),
+                })
+            );
+        } else {
+            if (storageConfig) {
+                sectionConfigs = sectionConfigs.push(
+                    Map({[FREEZERS_KEY]: storageConfig})
+                );
+            }
+            sectionConfigs = sectionConfigs.push(
+                Map({
+                    [WORKFLOW_KEY]: workflowConfig,
+                    [MEDIA_KEY]: getMediaSectionConfig(appBase),
+                    [NOTEBOOKS_KEY]: getNotebooksSectionConfig(appBase),
+                })
+            );
+        }
     } else {
-        const userSectionConfig = new MenuSectionConfig({
-            iconCls: 'fas fa-user-circle ',
-        });
-        sectionConfigs = sectionConfigs.push(Map<string, MenuSectionConfig>().set(USER_KEY, userSectionConfig));
+        if (storageConfig) {
+            sectionConfigs = sectionConfigs.push(Map({[FREEZERS_KEY]: storageConfig}));
+        }
+        sectionConfigs = sectionConfigs.push(Map ({[USER_KEY]: USER_SECTION_CONFIG}));
     }
     return sectionConfigs;
 }
 
-function getProductId(moduleName: string): string {
+export const useMenuSectionConfigs = (user: User, appProperties: AppProperties, moduleContext: any): List<Map<string, MenuSectionConfig>> => {
+    return useMemo(() => {
+        return getMenuSectionConfigs(user, appProperties.productId, moduleContext);
+    }, [user, moduleContext]);
+};
+
+function getProductId(moduleName: string, moduleContext: any): string {
     const lcModuleName = moduleName.toLowerCase();
-    const moduleContext = getServerContext().moduleContext[lcModuleName];
-    return moduleContext?.productId ? moduleContext.productId.toLowerCase() : undefined;
+    const context = (moduleContext ?? getServerContext().moduleContext)?.[lcModuleName];
+    return context?.productId?.toLowerCase();
 }
 
-function getApplicationUrlBase(moduleName: string, currentApp: string): string {
-    const appName = getProductId(moduleName);
-    return !appName || appName === currentApp.toLowerCase()
+function getApplicationUrlBase(moduleName: string, currentProductId: string, moduleContext: any): string {
+    const lcProductId = getProductId(moduleName, moduleContext);
+    return !lcProductId || lcProductId === currentProductId.toLowerCase()
         ? ''
-        : buildURL(appName, 'app.view', undefined, { returnUrl: false });
+        : buildURL(lcProductId, 'app.view', undefined, { returnUrl: false });
 }
 
 export function getDateFormat(): string {
     return getServerContext().container.formats.dateFormat;
 }
 
+// Returns the friendly name of the product, primarly for use in help text.
 export function getCurrentProductName() {
     const lcController = ActionURL.getController().toLowerCase();
     if (!lcController) return LABKEY_SERVER_PRODUCT_NAME;
```

Comment:
can we actually ever not have a controller name? seems fine to leave, I just am curious why this is needed (jest test case?).

Reply:
I wondered the same thing. Not sure when that happens except in the test environment.